sycl/include/sycl/detail/kernel_global_info.hpp

```diff
@@ -0,0 +1,23 @@
+//==-------------------- kernel_global_info.hpp -----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/detail/export.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+namespace free_function_info_map {
+
+__SYCL_EXPORT void add(const void *DeviceGlobalPtr, const char *UniqueId);
```

Comment:
This is information added for free functions specifically. It is okay to give specific arguments, like the number of arguments. Do we need the shim counter embedded?

Also, I would like the ID to be first, so it matches the idea of the map:
```suggestion
__SYCL_EXPORT void add(const char *UniqueId, unsigned NumArgs);
```
This should also be reflected in the following calls.

Reply:
I do not think we need embedded shim counter, UniqueId, i.e. kernel name string, is enough

Reply:
Just quickly unresolving this issue. I see you did the reordering, but missed the point about making the argument `unsigned` for `NumArgs`. I don't think there's a need for it to be converted to a `void *`. 🤔